app/modules/asset/service.server.ts

```diff
@@ -2824,3 +2824,29 @@ export async function relinkQrCode({
     }),
   ]);
 }
+
+export async function getAvailableAssetsForBooking(
+  assetIds: Asset["id"][]
+): Promise<string[]> {
+  try {
+    const selectedAssets = await db.asset.findMany({
+      where: { id: { in: assetIds }, status: "AVAILABLE" },
+      select: { status: true, id: true, kit: true },
+    });
+    if (selectedAssets.some((asset) => asset.kit)) {
+      throw new ShelfError({
+        cause: null,
+        message: "Cannot add assets that belong to a kit.",
+        label: "Booking",
+      });
+    }
+    return selectedAssets.map((asset) => asset.id);
+  } catch (err) {
```

Comment:
Please use `cause`. that is what we use everywhere. Also the message in the catch should be dynamic. something like this:
```ts
message: cause?.message ? cause.message : "some message"
```